clients/src/main/java/org/apache/kafka/common/requests/ShareGroupDescribeRequest.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.requests;
+
+import org.apache.kafka.common.message.ShareGroupDescribeRequestData;
+import org.apache.kafka.common.message.ShareGroupDescribeResponseData;
+import org.apache.kafka.common.protocol.ApiKeys;
+import org.apache.kafka.common.protocol.ByteBufferAccessor;
+import org.apache.kafka.common.protocol.Errors;
+
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class ShareGroupDescribeRequest extends AbstractRequest {
+
+  public static class Builder extends AbstractRequest.Builder<ShareGroupDescribeRequest> {
+
+    private final ShareGroupDescribeRequestData data;
+
+    public Builder(ShareGroupDescribeRequestData data) {
+      this(data, false);
+    }
+
+    public Builder(ShareGroupDescribeRequestData data, boolean enableUnstableLastVersion) {
+      super(ApiKeys.SHARE_GROUP_DESCRIBE, enableUnstableLastVersion);
+      this.data = data;
+    }
+
+    @Override
+    public ShareGroupDescribeRequest build(short version) {
+      return new ShareGroupDescribeRequest(data, version);
+    }
+
+    @Override
+    public String toString() {
+      return data.toString();
+    }
+  }
+
+  private final ShareGroupDescribeRequestData data;
+
+  public ShareGroupDescribeRequest(ShareGroupDescribeRequestData data, short version) {
+    super(ApiKeys.SHARE_GROUP_DESCRIBE, version);
+    this.data = data;
+  }
+
+  @Override
+  public ShareGroupDescribeResponse getErrorResponse(int throttleTimeMs, Throwable e) {
+    ShareGroupDescribeResponseData data = new ShareGroupDescribeResponseData()
+        .setThrottleTimeMs(throttleTimeMs);
+    // Set error for each group
+    this.data.groupIds().forEach(
+        groupId -> data.groups().add(
+            new ShareGroupDescribeResponseData.DescribedGroup()
+                .setGroupId(groupId)
+                .setErrorCode(Errors.forException(e).code())
```

Comment:
May be you want to compute the code once for all groups and use that.

Reply:
So the data has multiple share groups and we want to create the wrapper for each group. The group id would be different for each so we cannot reuse the computation.

Reply:
I think I mentioned `code` computation and not a wrapper, code computed once `Errors.forException(e).code()`, ofcourse if groupId would have been same then we wouldn't be iterating in the loop :)